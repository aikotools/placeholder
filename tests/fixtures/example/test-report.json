{
  "report": {
    "id": "{{gen:uuid:report-2025-001}}",
    "title": "E2E Test Report - Train System",
    "generated": "{{time:calc:0:yyyy-MM-dd\\:HH\\:mm\\:ss}}",
    "generatedTimestamp": "{{time:calc:0:seconds}}"
  },
  "testRun": {
    "runId": "{{gen:uuid:testrun-001}}",
    "startTime": "{{time:calc:0:seconds}}",
    "endTime": "{{time:calc:3600:seconds}}",
    "durationSeconds": "{{gen:number:3600}}",
    "startDate": "{{time:calc:0:dd.MM.yyyy}}",
    "startTimeFormatted": "{{time:calc:0:HH\\:mm\\:ss}}",
    "endTimeFormatted": "{{time:calc:3600:HH\\:mm\\:ss}}"
  },
  "statistics": {
    "totalTests": "{{gen:number:150}}",
    "passed": "{{gen:number:145}}",
    "failed": "{{gen:number:5}}",
    "skipped": "{{gen:number:0}}",
    "successRate": "{{gen:string:96.7|toNumber}}",
    "enabled": "{{gen:boolean:true}}"
  },
  "tests": [
    {
      "id": "{{gen:uuid:test-001}}",
      "name": "Train Departure Test",
      "status": "PASSED",
      "duration": "{{gen:number:120}}",
      "startTime": "{{time:calc:0:seconds}}",
      "endTime": "{{time:calc:120:seconds}}"
    },
    {
      "id": "{{gen:uuid:test-002}}",
      "name": "Train Arrival Test",
      "status": "PASSED",
      "duration": "{{gen:number:180}}",
      "startTime": "{{time:calc:120:seconds}}",
      "endTime": "{{time:calc:300:seconds}}"
    },
    {
      "id": "{{gen:uuid:test-003}}",
      "name": "Platform Assignment Test",
      "status": "FAILED",
      "duration": "{{gen:number:45}}",
      "startTime": "{{time:calc:300:seconds}}",
      "endTime": "{{time:calc:345:seconds}}"
    }
  ],
  "environment": {
    "name": "{{gen:string:test}}",
    "version": "{{gen:string:1.0.0}}",
    "build": "{{gen:number:42}}"
  },
  "artifacts": {
    "logFile": "test-run_{{time:calc:0:yyyy-MM-dd}}_{{gen:uuid:testrun-001}}.log",
    "screenshotDir": "screenshots/run_{{time:calc:0:yyyyMMdd}}_{{gen:string:001}}",
    "reportFile": "report_{{time:calc:0:yyyy-MM-dd\\:HH-mm-ss}}.html"
  }
}
