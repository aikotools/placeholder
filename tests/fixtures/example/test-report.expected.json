{
  "report": {
    "id": "report-2025-001",
    "title": "E2E Test Report - Train System",
    "generated": "2025-03-15:12:00:00",
    "generatedTimestamp": 1742040000
  },
  "testRun": {
    "runId": "testrun-001",
    "startTime": 1742040000,
    "endTime": 1742043600,
    "durationSeconds": 3600,
    "startDate": "15.03.2025",
    "startTimeFormatted": "12:00:00",
    "endTimeFormatted": "13:00:00"
  },
  "statistics": {
    "totalTests": 150,
    "passed": 145,
    "failed": 5,
    "skipped": 0,
    "successRate": 96.7,
    "enabled": true
  },
  "tests": [
    {
      "id": "test-001",
      "name": "Train Departure Test",
      "status": "PASSED",
      "duration": 120,
      "startTime": 1742040000,
      "endTime": 1742040120
    },
    {
      "id": "test-002",
      "name": "Train Arrival Test",
      "status": "PASSED",
      "duration": 180,
      "startTime": 1742040120,
      "endTime": 1742040300
    },
    {
      "id": "test-003",
      "name": "Platform Assignment Test",
      "status": "FAILED",
      "duration": 45,
      "startTime": 1742040300,
      "endTime": 1742040345
    }
  ],
  "environment": {
    "name": "test",
    "version": "1.0.0",
    "build": 42
  },
  "artifacts": {
    "logFile": "test-run_2025-03-15_testrun-001.log",
    "screenshotDir": "screenshots/run_20250315_001",
    "reportFile": "report_2025-03-15:12-00-00.html"
  }
}
